.github/workflows/rust.yml

```diff
@@ -51,6 +51,34 @@ jobs:
       - name: Run tests
         run: |
           cargo test --all-features
+  linux-arm:
+    runs-on: buildjet-4vcpu-ubuntu-2204-arm
+    timeout-minutes: 30
+    defaults:
+      run:
+        working-directory: ./rust
+    steps:
+      - uses: actions/checkout@v3
+      - uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: rust
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y protobuf-compiler libssl-dev pkg-config
+          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+      - name: Run cargo fmt
+        run: cargo fmt --check
+      # split everything us so we know what's slow.
+      - name: Build
+        run: |
+          cargo build --no-default-features
+      - name: Build test
+        run: |
+          cargo test --no-run
```

Comment:
I think we can just drop this. It's recompiling the test binaries because `CARGO_INCREMENTAL=0`, I think.

```suggestion
```